### PR TITLE
Small fixes, config good to go.

### PR DESCRIPTION
**Pull request by ProgrammerDan**
Should be safe, give it a onceover. Includes line 10k fixes.

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -7905,4 +8203,2746 @@ production_recipes:
...
+        material: IRON_CHESTPLATE
+        amount: 1
+      Emerald Blocks:
+        material: EMERALD_BLOCK
+        amount: 2
+    outputs:
+      Enchanted Iron Chestplate:
+        material: IRON_CHESTPLATE
+        amount: 1
+    enchantments:
```
Comment (ProgrammerDan) [inline config.yml]: Don't merge, broken line.

Comment (ProgrammerDan) [inline config.yml]: ok to merge, fixed.

**Comment by ttk2** [comment]
ok to test

**Comment by ttk2** [comment]
testing build and config on Civtest, will merge if your happy with it, thanks!

**Comment by rourke750** [comment]
ok to test

**Comment by ProgrammerDan** [comment]
@ttk2 config needs more tweak, but I wanted to get the fixes merge for
saturday testers.

Balance config per your orders will be up soon, I will ping you again then

**Comment by ttk2** [comment]
ok move into testing on Discourse then?

**Comment by ProgrammerDan** [comment]
Getting reports that enchantment factories aren't working as advertised, looking into it

**Comment by DoctorCelebro** [comment]
1.8 added an extra enchantment on boots called "Depth Strider". Too late to add that in I suppose?

http://minecraft.gamepedia.com/Enchanting#Enchantments

**Comment by ttk2** [comment]
we can tack it on later.

**Comment by ProgrammerDan** [comment]
@dbear20 you need to sign the CLA as soon as you can.

**Comment by ttk2** [comment]
just click the details button the clahub line above the big merge with caution stuff. Only takes a second.

**Comment by dbear20** [comment]
Sorry about that I forgot that was added. Is there one of those for each repo?

**Comment by ttk2** [comment]
I think so now.

**Comment by ttk2** [comment]
anyways is this ready to merge now?

**Comment by BlackXnt** [comment]
There is still a problem with the random enchanting factories.

**Comment by ProgrammerDan** [comment]
@ttk2 unless you or matta has an objection, I'm going to merge the repair and general compactor into this branch. I've got to go out with the family for a few hours, but if there are no "don't do it" 's by then I'll merge it in. 

Quick notes:

Reset Factory costs have been updated to the discussed values.

Compactor costs have been updated to the discussed values.

Compactor looks good. I was a little worried about book duping, but so far haven't found a way to do it, nor any other item. It compacts every non-lored item just fine it seems, and decompacts nicely too. 

Only problem I've found is there is a duplicating bug if MC copying is enabled. You can make normal copies of a Compacted stack of signed books; this would crash the book sales market. I think I'll just have to add an "excluded" config and any items we find that can be duped, we'll add it there and it just won't be compactable.

**Comment by BlackXnt** [comment]
> Reset Factory costs have been updated to the discussed values. Compactor costs have been updated to the discussed values.

What are the discussed values?

**Comment by ProgrammerDan** [comment]
https://github.com/ProgrammerDan/FactoryMod/tree/fixes-repair-2

Basically, in the balance thread TTK issued mandate that price to reset a single tool is 1.5x cost of 1.7 repair. I used model of high-tier named tool to figure out difference in cost of 5 repairs 1.7 vs. 5 repairs 1.8, took 1.7's cost times 1.5 and subtracted 1.8's cost; this leads to per-reset cost. Factory cost is increased to compensate for the known reduction in overall repair costs.

**Comment by ttk2** [comment]
we might increase the run cost later, so don't try too hard to balance it out with startup, but also don't worry about it in general too much, need this online rather than fettering over it.

**Comment by ProgrammerDan** [comment]
Yeah, these have to be final costs for reset and compactor until after its
live.

My burn list is:

* prevent compacting of items with NBT data to prevent duplication of
books. Failing that, add specific exclusion list.

* ensure minimum 1 enchant from random enchant factories with new config
flag in production recipes.

* briefly check if feasible to leverage bucket enchantment safety checks to
overcome objectionable random combinations.

* merge repair and compactor into mattas recipes as one pull

* get testers on this asap

**Can release at this point.** General compactor needs to stay locked until
we can make sure edge cased are tested.

Post release:

* replace nether compactor items with general compactor items (just nether
crates * 16)

* remove nether compactor

* fixes and changes

* get testers

Second release can then go out.

**Comment by ttk2** [comment]
sounds good, I am kinda surprised just how much work this ended up being, but I guess the compactor is non-trivial.

**Comment by ProgrammerDan** [comment]
ehhh, these two weeks have all been subtle, hidden bugs. Future factories if we leverage factorymod will take less time to debug, I'm far more familiar with it now.

The issue was just that the base code for Reset and Compactor had a few significant flaws, and some of the flaws actively hid other flaws ... so ... just a day-in-the-life-of for software development. There is almost always overrun on testing.

**Comment by ProgrammerDan** [comment]
Another fun dup: you can cut the cost of applying dyes to banners by 16x by compacting a blank banner, then applying the dyes, then decompacting. Pretty fun! You can also simply duplicate the compacted banner and the lore is preserved, so you can get effectively infinite banners.

I'm probably going to have to go the route of exclusion, there are just a lot of things that can go wrong here.

**Comment by ttk2** [comment]
That's fine for now. If someone wants to fix it then can tackle it.

**Comment by ProgrammerDan** [comment]
Ok quick update. 

Known dups are all conquered. Basically any item with a "Complex" metadata block will not be considered compactable. It's a bit hackish but works beautifully and in a single check with no new config I can block compaction of books, banners, fireworks, etc.

Enchanting fixes will be easy as well, everything we want is just built into Bukkit. I just have to modify it a bit, got started way too late last night and needed sleep. 

So I'll be merging the repair and compactor into this branch tonight, then it's a go for testing. 

Note, compactor will stay locked, even though I'm 99% certain it's good to go now. Will take me a while to convert the nether compactor recipes to regular compactor recipes, and I don't want that to hold things up further.

**Comment by ttk2** [comment]
sounds great, hopefully we can finally put this behind us, or at least online, by the end of this week.

**Comment by ttk2** [comment]
should I take the latest version of this and put it into testing?

**Comment by ProgrammerDan** [comment]
I should have a bunch of new commits shortly, wait for that. Then we can have a full day of testing tomorrow. I'll start a new Testing thread, the old one is of limited usefulness now.

**Comment by ProgrammerDan** [comment]
Just some last local tests but tons of changes about to hit this branch.

**Comment by ProgrammerDan** [comment]
@ttk2 @rourke750 @drjawa 

Not sure who is still awake, but this is ready for civtest. Need to take build jar and config; tons of changes, improvements, and fixes across the board. I've tested the bulk of them locally but this absolutely needs a full day of aggressive testing, however we can push it we should. I'm going to take whatever energy/time left in my evening and write up a new testing thread for all these fixes and changes.

This accomplishes everything I have on my plate for pre-release, pending any new identified bugfixes.

I'll leave my plate open for bugfixes, then I'll start in on Release 2 items.

**Comment by drjawa** [comment]
I put this (http://build.civcraft.co:8080/job/FactoryMod-pull/22/) on civtest.

Will merge if all goes well.

**Comment by ProgrammerDan** [comment]
spongesuds reports no Bane on deterministic sword enchantment factory -- @Goldmattress is this intentional?

**Comment by ttk2** [comment]
Is a new testing thread made for this? Great work Dan be sure to link to
the testing thread form Reddit and the like.

**Comment by ProgrammerDan** [comment]
Jawa bumped the original testing thread and I updated OP

**Comment by ProgrammerDan** [comment]
@Goldmattress QQ: 90% repair cost for deterministic factories? Don't recall if you ever stated if that was intentional or a mistake. 


ref:
https://github.com/Civcraft/FactoryMod/pull/106/files#diff-0f1be8eec416e71c28b41efb0544df26R1223

**Comment by hlprimm** [comment]
Are the XP recipes in this pull going on civcraft?

**Comment by ProgrammerDan** [comment]
@hlprimm  Yes

**Comment by ProgrammerDan** [comment]
Note to self:

* Ensure bane is gone from Deterministic Sword factory.

* Change diamond / emerald *blocks* to just plain diamonds and emeralds for repair costs on New Factories

* Make cost for Nether Compactor/Decompactor unique (compactor put to 70d, decompactor 130d).

* Push update to here

@Goldmattress -- am I missing anything above?

Then

* Investigate compactor infinite dup bug, and redstone activation bug

**Comment by DoctorCelebro** [comment]
Hmm...any idea why emerald cauldron requires the sand and the diamond requires bottles(Emerald Cauldron requires 8,192 sand, Diamond Cauldron requires 8,192 bottles)? If you take the glass smelter into effect, you are basically paying almost 3 times more bottles for only 20 blocks more. I recommend either switch the sand to bottles or divide the number of required sand by 3.

**Comment by ProgrammerDan** [comment]
Changes in newest commit:

* Bane enchantment removed from Sword Enchantment Factory, recipe removed from config as well.

* Nether Compactor set to 70d install
* Nether Decompactor set to 130d install
* Iron Weapon Factory set to 3 Iron Blocks repair (10% of 32 Iron Blocks to install)
* Random Diamond Tool Factory set to 9 Emerald Blocks repair (10% of 96 E Block to install)
* Random Diamond Armor Factory set to 6 Emerald Blocks repair (10% of 64 E Block to install)
* Random Diamond Weapon Factory set to 3 Emerald Blocks repair (10% of 32 E Block to install)
* Sword Enchantment Factory set to 30 Diamonds and 60 Emeralds to repair (10% of 32 D Blocks and 64 E Blocks to install)
* Armour Enchantment Factory set to 60 Diamonds and 90 Emeralds to repair (10% of 64 D Blocks and 96 E Blocks to install)
* Tool Enchantment Factory set to 90 Diamonds and 128 Emeralds to repair (10% of 96 D Blocks and 128 E Blocks to install)

* NETHER_WART was somehow still in the Wood_XP_1 recipe, altered to NETHER_STALK.

**Comment by ttk2** [comment]
I can just grab a new config for Civtest right?

**Comment by drjawa** [comment]
Yup, the only changes were to the config.

**Comment by ttk2** [comment]
ok to test

**Comment by ttk2** [comment]
on civtest

**Comment by ProgrammerDan** [comment]
@DoctorCelebro Matta clarified, altered to 32 crates of glass bottles now.

**Comment by ProgrammerDan** [comment]
@ttk2 Please put latest config on and restart Civtest. Current config is old config.

**Comment by ProgrammerDan** [comment]
Few label fixes. I ran every XP recipe, and compacted/decompacted every recipe ingredient using the nether compactor/decompactor. I also confirmed via the ingredients match function that the crates as produced in the nether compactor, decompactor fully satisfy the needs of the Emerald Cauldron.

**Comment by ttk2** [comment]
updated on Civtest, announcing final day of testing and staging for tomorrow mornings restart

File: config.yml [hunk 2]
```diff
@@ -7905,4 +8157,2805 @@ production_recipes:
...
+        probability: 0.02
+      Efficiency 3:
+        type: DIG_SPEED
+        level: 3
+        probability: 0.02
+      Fortune 2:
+        type: LOOT_BONUS_BLOCKS
+        level: 2
+        probability: 0.36
+  Create_Wooden_Axe:
```
Comment (ProgrammerDan) [inline config.yml]: Wooden Axe appears twice.

Replace the first with "Wooden Pickaxe"

**Comment by WildWeazel** [comment]
Let's be sure that people testing are also checking their factories from prior days, not just creating new ones.

**Comment by ProgrammerDan** [comment]
FWIW I retested all the factories I and BlackXnt created from early on in testing last night while giving someone a tour of the new enchanting system on civtest, they all worked great.

**Comment by ProgrammerDan** [comment]
@ttk2 Yeah, putting these fixes on Civtest & restart would be ideal -- only config changes this time

**Comment by ttk2** [comment]
on civtest, I am also staging this config for restart

**Comment by MonsieurWTF** [comment]
So what's going on with the cauldron recipes, did you pick one already or?

**Comment by ttk2** [comment]
feedback several days ago was really negative when I tried to select yours, probably going to go with Matta's to get something out the door and then try and evaluate where yours went wrong.

**Comment by MonsieurWTF** [comment]
It went wrong in primarily dirt at the entrance and some minor grievance on the lava count, along with a lot of bad mathematical calculations that helped fuel the outburst.

Both were properly reduced and switched to reduce impact on severe non-renewable materials.

I've just been waiting for a response from anyone on the Dev end about the changes since there were not any further gripes on what to do about it.

**Comment by ttk2** [comment]
this isn't the place to have this discussion in full, bump your discourse thread please.

**Comment by ProgrammerDan** [comment]
Just in general too late to back out, we've just gotta get something online in whole. This has been tested and vetted 100% besides the couple enchantment factories left.

**Comment by MonsieurWTF** [comment]
Considering it would hardly take more than 1-2 hours to test every recipe on my end and have it replace the Cauldron recipes present, the only thing waiting on it is an as stated above civtest and "Merge/No merge".

**Comment by ProgrammerDan** [comment]
If we keep trying to go back and forth on this, nothing will ever get out. Config change will be two days of additional testing, minimum, likely much more.

**Comment by ttk2** [comment]
this is whats going online, that's pretty much final, in the future we need to be more on top of these things sooner

**Comment by hlprimm** [comment]
Everything is good to go test-wise. 

I say announce tomorrow that it's happening Friday. It'll hit people out of nowhere if you stick it online all of a sudden tomorrow without any kind of warning on reddit. But that's just my two cents =P

**Comment by drjawa** [comment]
We can update the documentation Thursday and hopefully put it live Friday.

**Comment by ttk2** [comment]
Um like I said yesterday its staged to go up today. A little late to try
and much about it.

**Comment by drjawa** [comment]
My sense of time is a bit distorted.

**Comment by drjawa** [comment]
Now that this is properly live shouldn't this be merged into master?

**Comment by MonsieurWTF** [comment]
If this is merged I can get started tonight on fixing whatever gaps were left between the General Compactor and Emerald Cauldron to allow the general crates to feed into it.

**Comment by ProgrammerDan** [comment]
@BlackXnt sign the CLA and we are GTG

**Comment by BlackXnt** [comment]
Done. It might be a good idea to not wait for the daily restart and take down the server now for a few mins to upload the new jar.

**Comment by ProgrammerDan** [comment]
Note: #109 solved, thanks for being my proxy, BlackXnt!

@ttk2 as Black is saying this is good to go for install and restart.

**Comment by ttk2** [comment]
ok to test

thanks dan! going to test and stage for tomorrow mronings restart,  I would hate to ruin the planned nether party :P

**Comment by ttk2** [comment]
ok to test

**Comment by ttk2** [comment]
can't have anything else in the testing comment it seems.

**Comment by ttk2** [comment]
I think this is good for a merge now, any objections?

**Comment by rourke750** [comment]
Lol we should of had those commits squashed, oh well.

**Comment by ProgrammerDan** [comment]
Yeah, my bad ...

**Pull request merged**